tests/unit/test_workflow.py

```diff
@@ -442,3 +442,62 @@ def test_chaining_3():
     assert all(
         x in result.columns for x in ["ad_id_count", "ad_id_clicked_sum_ctr", "ad_id_clicked_sum"]
     )
+
+
+@pytest.mark.parametrize("shuffle", [nvt.io.Shuffle.PER_WORKER, nvt.io.Shuffle.PER_PARTITION, None])
+@pytest.mark.parametrize("use_client", [True, False])
+def test_workflow_apply(client, use_client, tmpdir, shuffle):
```

Comment:
Maybe add an `apply_online` parameter here to test the online case.

Reply:
My bad, it was not being covered. I have added the parameter, and I am running into an error that seems unrelated to this dtypes issue. 

 Digging into that I have found that [when calling to ](https://github.com/NVIDIA/NVTabular/blob/main/nvtabular/workflow.py#L816) `dataset.to_iter` no columns are passed, which it looks like it will never work for me.

Reply:
@rjzamora have we notices these problem before?

Reply:
>@rjzamora have we noticed this problem before?

Good catch!! The code you referenced above should probably be changed to:

```python
        # Iterate through dataset, apply ops, and write out processed data
        if apply_ops:
            columns = self.columns_ctx["all"]["base"]
            for gdf in dataset.to_iter(shuffle=(shuffle is not None), columns=columns):
                self.apply_ops(gdf, output_path=output_path, writer=writer, dtypes=dtypes)
```

Along with [my other suggestion](https://github.com/NVIDIA/NVTabular/pull/392/files#r516341260) - The tests *should* pass.

Reply:
Got it now!